element_facemap/facemap_train.py

```diff
@@ -0,0 +1,387 @@
+    assert hasattr(
+        linking_module, "get_facemap_root_data_dir"
+    ), "The linking module must specify a lookup function for a root data directory"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    # activate facial behavioral extimation (fbe) schema
+    fbe.activate(
+        fbe_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+
+    # activate facial pose model schema
+    facemap_inference.activate(
+        facemap_model_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+
+    # activate facemap train schema
+    schema.activate(
+        facemap_train_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+# ----------------------------- Table declarations ----------------------
+
+
+@schema
+class FacemapTrainFileSet(dj.Manual):
+    """Collection of files associated with a given training set.
+
+    Attributes:
+        file_set_id (int): Unique ID for each collection of training files."""
+
+    definition = """ # Set of vids in training set
+    file_set_id: int
+    """
+
+    class VideoFile(dj.Part):
+        """Video File IDs and paths in a given FacemapTrainFileSet
+
+        Attributes:
+            FacemapTrainFileSet (foreign key)   : FacemapTrainFileSet key.
+            video_file_id (int)                 : Video File index
+            video_file_path ( varchar(255) )    : Path to file on disk relative to root."""
+
+        definition = """ # Paths of training files (e.g., .avi, .mp4 video files)
+        -> master
+        video_file_id: int
+        ---
+        video_file_path: varchar(255)
+        """
+    
+    class KeypointsFile(dj.Part):
+        """Keypoints File containing labels and paths in a given FacemapTrainFileSet
+
+        Attributes:
+            FacemapTrainFileSet (foreign key)   : FacemapTrainFileSet key.
+            file_id                             : Keypoint File index.  
+            file_path ( varchar(255) )          : Path to file on disk relative to root."""
+
+        definition = """ # Paths of training files (e.g.: .h5 keypoints data file)
+        -> master
+        ---
+        file_path: varchar(255)
+        """
+
+
+@schema
+class FacemapTrainParamSet(dj.Lookup):
+    """Parameters used to train a model
+
+    Attributes:
+        paramset_idx (smallint): Index uniqely identifying each paramset.
+        paramset_desc ( varchar(128) ): Description of paramset.
+        param_set_hash (uuid): Hash identifying this paramset.
+        params (longblob): Dictionary of all applicable parameters.
+        Note: param_set_hash must be unique."""
+
+    definition = """
+    # Parameters to specify a facemap model training instance
+    paramset_idx                  : smallint
```

Comment:
Smallint is better if we know that the integer won't be larger than a certain size, since smallints take 2 bytes per integer while ints take 4 bytes per integer. Should reduces the load on the database.

Reply:
Yep, I think that's totally fine. Let's go with it. We can consider this optimization across other Elements as well